File: android-client/app/src/main/java/com/example/android_client/core/network/AuthDtos.kt
package com.example.android_client.core.network

import kotlinx.serialization.Serializable

@Serializable
data class LoginRequest(val email: String, val password: String)

@Serializable
data class LoginResponse(val token: String, val refreshToken: String)

@Serializable
data class RefreshTokenRequest(val refreshToken: String)

File: windows-client/src/Hikari.WindowsClient/Core/Network/AuthDtos.cs
namespace Hikari.WindowsClient.Core.Network;

public sealed class LoginRequest
{
    public string Username { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;

    public LoginRequest() { }

    public LoginRequest(string username, string password)
    {
        Username = username;
        Password = password;
    }
}

public sealed class LoginResponse
{
    public string Token { get; set; } = string.Empty;
    public string RefreshToken { get; set; } = string.Empty;
}

public sealed class RefreshTokenRequest
{
    public string RefreshToken { get; set; } = string.Empty;

    public RefreshTokenRequest() { }

    public RefreshTokenRequest(string refreshToken) => RefreshToken = refreshToken;
}
